pkg/tasks/controlplane.go

```diff
@@ -64,6 +65,7 @@ func (t *ControlPlaneTask) Run(ctx context.Context) error {
 	}
 
 	if t.config.ClusterMonitoringConfiguration.K8sPrometheusAdapter.DedicatedServiceMonitors.Enabled {
+		klog.V(4).Info("Dedicated ServiceMonitors are deprecated, this setting has no effect in this version and will be removed in a future release")
```

Comment:
we'd need to remove the service monitor?

Reply:
I left it in for now since a user might actually use the provided metric outside of just prometheus adapter (it has its own name). So removing the SM would be something for next release?

Or should we refer to our lack of guarantee for metric name stability?

Reply:
I'd be fine with dropping the metrics given we provide no guarantee.

Reply:
I think we can go further and delete the related resources from `assets/` to avoid any confusion and just use their names to delete them.
(add a TODO remove the delete code or even a ticket to remove it in future releases will be great)

Reply:
Added the the resource deletion.